src/test/javascript/spec/component/programming-exercise/test-case-distribution-chart.component.spec.ts

```diff
@@ -0,0 +1,167 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { TranslateService } from '@ngx-translate/core';
+import { TestCaseDistributionChartComponent } from 'app/exercises/programming/manage/grading/charts/test-case-distribution-chart.component';
+import { MockModule, MockPipe } from 'ng-mocks';
+import { BarChartModule } from '@swimlane/ngx-charts';
+import { ProgrammingExerciseTestCase, Visibility } from 'app/entities/programming-exercise-test-case.model';
+import { ProgrammingExercise } from 'app/entities/programming-exercise.model';
+import { ArtemisTranslatePipe } from 'app/shared/pipes/artemis-translate.pipe';
+import { TestCaseStatsMap } from 'app/entities/programming-exercise-test-case-statistics.model';
+
+describe('Test case distribution chart', () => {
+    const programmingExercise = new ProgrammingExercise(undefined, undefined);
+    let component: TestCaseDistributionChartComponent;
+    let fixture: ComponentFixture<TestCaseDistributionChartComponent>;
+
+    const configureComponent = (testCases: ProgrammingExerciseTestCase[]) => {
+        configureProgrammingExercise();
+        component.testCases = testCases;
+    };
+    const configureProgrammingExercise = () => {
+        programmingExercise.maxPoints = 10;
+        programmingExercise.bonusPoints = 0;
+        component.exercise = programmingExercise;
+    };
+
+    const testCase = {
+        id: 1,
+        visibility: Visibility.Always,
+        weight: 1,
+        exercise: programmingExercise,
+        bonusPoints: 0,
+        bonusMultiplier: 1,
+        testName: 'test case',
+    } as ProgrammingExerciseTestCase;
+
+    const testCase2 = {
+        id: 2,
+        visibility: Visibility.Always,
+        weight: 1.5,
+        exercise: programmingExercise,
+        bonusPoints: 0,
+        bonusMultiplier: 1,
+        testName: 'test case 2',
+    } as ProgrammingExerciseTestCase;
+
+    const invisileTestCase = {
+        id: 3,
+        visibility: Visibility.Never,
+        weight: 4,
+        exercise: programmingExercise,
+        bonusPoints: 1,
+        bonusMultiplier: 1,
+        testName: 'invisible test case',
+    } as ProgrammingExerciseTestCase;
+
+    const testCase4 = {
+        id: 4,
+        visibility: Visibility.AfterDueDate,
+        weight: 2,
+        exercise: programmingExercise,
+        bonusPoints: 2,
+        bonusMultiplier: 1,
+        testName: 'test case 4',
+    } as ProgrammingExerciseTestCase;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule, MockModule(BarChartModule)],
+            declarations: [TestCaseDistributionChartComponent, MockPipe(ArtemisTranslatePipe)],
+            providers: [{ provide: TranslateService, useClass: MockTranslateService }],
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(TestCaseDistributionChartComponent);
+        component = fixture.componentInstance;
+    });
```

Comment:
```suggestion
    });

```
Missing newline before new test case.